test/rtl/canonicalization.mlir

```diff
@@ -18,6 +20,17 @@ func @extract_cstfold() -> i3 {
   return %x : i3
 }
 
+// CHECK-LABEL: func @and_cstfold() -> i7 {
+// CHECK-NEXT:    %c1_i7 = rtl.constant(1 : i7)
+// CHECK-NEXT:    return %c1_i7
+
+func @and_cstfold() -> i7 {
+  %c11_i7 = rtl.constant(11 : i7) : i7
+  %c5_i7 = rtl.constant(5 : i7) : i7
+  %x = rtl.and %c11_i7, %c5_i7 : i7
```

Comment:
Please add one more non-constant operand, so this is something like "and(x, 11, 5)".  This will make sure to test your path, instead of the "All constant" path implemented in the fold hook.

Reply:
Done.